contracts/EscrowAndStaking/StakingThales.sol

```diff
@@ -206,6 +206,43 @@ contract StakingThales is IStakingThales, Initializable, ProxyOwned, ProxyReentr
     function getSNXStaked(address account) external view returns (uint) {
         return _getSNXStakedForAccount(account);
     }
+    
+    function getBaseReward(address account) public view returns (uint) {
+        return _stakedBalances[account]
+                .add(iEscrowThales.getStakedEscrowedBalanceForRewards(account))
+                .mul(currentPeriodRewards)
+                .div(_totalStakedAmount.add(_totalEscrowedAmount));
+    }
+    
+    function getAMMVolume(address account) external view returns (uint) {
+        return _getTotalAMMVolume(account);
+    }
+    
+    function getSNXBonus(address account) public view returns (uint) {
+        uint baseReward = getBaseReward(account);
+        uint stakedSNX = _getSNXStakedForAccount(account);
+        // SNX staked more than base reward
+        return stakedSNX >= baseReward
+            ? 15
+            : stakedSNX.mul(15).div(baseReward);
+    }
+    
+    function getAMMBonus(address account) public view returns (uint) {
+        uint baseReward = getBaseReward(account);
+        return _getTotalAMMVolume(account) >= baseReward.mul(10)
+            ? 12
```

Comment:
can this be some kind of property or it is fix

Reply:
fast code, it's a view. 
The main code is in another function

Reply:
Cool